src/os/bluestore/BlueStore.cc

```diff
@@ -6322,11 +6323,19 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 		   << dendl;
 	} else {
 	  _txc_finalize_kv(txc, txc->t);
+	  if (g_conf->bluestore_rtc && !txc->is_pipelined_io && txc->wal_txn) {
+	    if (txc->osr->kv_finisher_submitting == 1) {
+	      db->submit_transaction_sync(txc->t);
+	      _txc_state_proc(txc);
+	      return;
+	    }
+	  }
 	  txc->kv_submitted = true;
```

Comment:
Actually, no.. i think we want a different flag, kv_submitted_sync, so that we can explicitly condition the sycn completions on a sync commit of the transaction (because it is strictly dependent on being in the caller context holding pg lock).